go-runtime/ftl/singleton.go

```diff
@@ -0,0 +1,31 @@
+package ftl
+
+import (
+	"context"
+	"sync"
+)
+
+type InputLambda[T any] func(context.Context) (T, error)
```

Comment:
Let's not worry about the intermediate type - the signature is so simple that I think it just adds noise rather than reducing it.

Reply:
Sounds good, removed